subprojects/docs/src/samples/user-input/pass-credentials-to-external-tool-via-stdin/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,13 @@
+val login = tasks.register<Exec>("login") {
+    val username: String by project
```

Comment:
I think we want to try to use the `ProviderFactory` APIs here so that IE captures these as build inputs

Reply:
Would it make sense to update this section of the documentation to favor `ProviderFactory` as well? https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties